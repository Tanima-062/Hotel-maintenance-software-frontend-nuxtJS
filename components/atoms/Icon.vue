<template>
  <div :class="styles">
    <svg
      class="w-7 h-7 fill-current"
      :class="`${color}`"
      :style="`width: ${iconWidth}px; height: ${iconHeight}px`"
      :viewBox="`0 0 24 24`"
    >
      <path :d="icon" />
    </svg>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  icon!: string

  @Prop({ default: 'text-white' })
  color!: string

  @Prop({ default: '' })
  bgColor!: string

  @Prop({ default: false })
  round!: boolean

  @Prop({ default: 24 })
  iconWidth!: number

  @Prop({ default: 24 })
  iconHeight!: number

  @Prop({ default: '' })
  extraClass!: string

  get styles(): string {
    const classList = []
    if (this.bgColor) {
      classList.push(`${this.bgColor}`)
    }
    if (this.round) {
      classList.push('rounded-full')
    }
    if (this.extraClass) {
      classList.push(this.extraClass)
    }
    return classList.join(' ')
  }
}
</script>
