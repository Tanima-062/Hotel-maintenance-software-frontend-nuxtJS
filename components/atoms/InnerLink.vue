<template>
  <nuxt-link
    class="cursor-pointer focus:outline-none"
    :class="styles"
    :to="href"
  >
    <slot />
  </nuxt-link>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  href!: string

  @Prop({ default: 'default' })
  textColor!: string

  @Prop({ default: 'default' })
  hoverTextColor!: string

  @Prop({ default: true })
  showUnderline!: boolean

  @Prop({ default: true })
  focusOn!: boolean

  get styles(): string {
    const classList = []
    if (this.showUnderline) {
      classList.push('underline')
    }
    classList.push(`${this.textColor}`)
    classList.push(`hover:${this.hoverTextColor}`)
    classList.push(this.focusOn ? 'focus:shadow-outline' : 'focus:shadow-none')
    return classList.join(' ')
  }
}
</script>
