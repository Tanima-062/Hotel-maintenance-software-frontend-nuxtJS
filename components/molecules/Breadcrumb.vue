<template>
  <div class="flex items-end">
    <div
      v-for="(item, key) in breadcrumbList"
      :key="key"
      class="flex items-center"
    >
      <p v-if="key === 0" class="whitespace-no-wrap text-black">
        {{ item.name }}
      </p>

      <InnerLink
        v-else-if="breadcrumbList.length - 1 !== key"
        class="ml-3"
        text-color="text-black"
        :href="item.href"
        :show-underline="false"
        :focus-on="false"
      >
        {{ item.name }}
      </InnerLink>

      <p
        v-else
        class="whitespace-no-wrap ml-3 text-3xl text-black-dark font-bold"
      >
        {{ breadcrumbList[breadcrumbList.length - 1].name }}
      </p>
      <Icon
        v-if="breadcrumbList.length - 1 !== key"
        :icon="chevronRight"
        class="mt-1 ml-3"
        color="text-black"
      />
    </div>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'
import { mdiChevronRight } from '@mdi/js'

@Component({
  components: {
    InnerLink: () => import('~/components/atoms/InnerLink.vue'),
    Icon: () => import('~/components/atoms/Icon.vue'),
  },
})
export default class extends Vue {
  @Prop({ default: [], required: true })
  breadcrumbList!: string[]

  private chevronRight = mdiChevronRight
}
</script>
