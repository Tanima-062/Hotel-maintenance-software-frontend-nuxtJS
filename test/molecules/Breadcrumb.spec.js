import { mount } from '@vue/test-utils'
import Breadcrumb from '@/components/molecules/Breadcrumb.vue'
import Icon from '@/components/atoms/Icon.vue'
import InnerLink from '@/components/atoms/InnerLink.vue'

describe('Molecules Breadcrumb Check', () => {
  test('Breadcrumb Test', () => {
    const props = {
      breadcrumbList: [
        { name: 'bread1', href: '/' },
        { name: 'bread2', href: '/' },
      ],
    }
    const wrapper = mount(Breadcrumb, {
      propsData: props,
      stubs: {
        Icon,
        InnerLink,
      },
    })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
