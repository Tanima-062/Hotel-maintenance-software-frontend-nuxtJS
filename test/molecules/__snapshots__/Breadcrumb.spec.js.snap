// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules Breadcrumb Check Breadcrumb Test 1`] = `
<div
  class="flex items-end"
>
  <div
    class="flex items-center"
  >
    <p
      class="whitespace-no-wrap text-black"
    >
      
      bread1
    
    </p>
     
    <div
      class="mt-1 ml-3"
    >
      <svg
        class="w-7 h-7 fill-current text-black"
        style="width: 24px; height: 24px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
        />
      </svg>
    </div>
  </div>
  <div
    class="flex items-center"
  >
    <p
      class="whitespace-no-wrap ml-3 text-3xl text-black-dark font-bold"
    >
      
      bread2
    
    </p>
     
    <!---->
  </div>
</div>
`;
